Verb confirmation

ShadowCommander opened the pull request:
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->

Adds a confirmation button for more destructive verbs.

**Screenshots**
<!-- If applicable, add screenshots to showcase your PR. If your PR is a visual change, add
screenshots or it's liable to be closed by maintainers. -->

https://user-images.githubusercontent.com/10494922/149106430-83f3f76a-5b62-4c61-8ac1-951958eb507c.mp4

ElectroJr commented on Content.Client/Verbs/UI/ConfirmationMenuElement.cs:
```diff
@@ -0,0 +1,33 @@
+    public ConfirmationMenuElement(Verb verb, string? text, VerbType type) : base(text)
+    {
+        Verb = verb;
+        Type = type;
+
```
```suggestion
        Type = type;
        Icon.Visible = false;

```
Or slap a question mark or warning/exclamation mark in there or something. Currently its just an empty space.

ElectroJr replied:
Though obviously if you decide to replace the expansion-indicator icon with a symbol, then this should just be invisible and vice versa

ElectroJr commented on Content.Client/Verbs/UI/VerbMenuPresenter.cs:
```diff
@@ -170,7 +170,13 @@ public override void OnKeyBindDown(ContextMenuElement element, GUIBoundKeyEventA
 
             if (element is not VerbMenuElement verbElement)
+            {
+                if (element is not ConfirmationMenuElement confElement)
+                    return;
+
+                ExecuteVerb(confElement.Verb, confElement.Type);
```
IMO it would be a bit easier to make sense of the code if you move this outside of the if block and just make it
```cs
if (element is ConfirmationMenuElement confElement)
{
    ExecuteVerb(confElement.Verb, confElement.Type);
    return;
}
```

ShadowCommander replied:
I made it this way because the confirmation menu is rare with only three verbs using it at the moment.

ElectroJr replied:
fair point.

ElectroJr commented on Content.Client/Verbs/UI/VerbMenuPresenter.cs:
```diff
@@ -196,7 +202,26 @@ public override void OnKeyBindDown(ContextMenuElement element, GUIBoundKeyEventA
             }
 
-            _verbSystem.ExecuteVerb(CurrentTarget, verb, verbElement.Type);
+            if (verb.ConfirmationPopup)
```
Not directly related to this PR, but there's a comment above this that was just never updating during the last PR that just misleads/confuses people now. 
```
// We will act as if they clicked on the first verb in that category.
```

Maybe just remove it or update it in this pr?

ElectroJr reviewed (commented):
There is one bit of jank, where because the sub-menu is only ever added when the verb is clicked, it gets given an "expansion indicator" after having being clicked and the sub-menu actuall opens on on-top of the other verb. if confirmation-verbs are only ever for admin use I guess this isn't all that important.

https://user-images.githubusercontent.com/60421075/149152933-16a8c862-9501-4f5e-8520-2fdacf8679e1.mp4

`ExpansionIndicator.Visible` is set in the `ContextMenuElement.SubMenu` setter.

So possible fixes:
- Move it outside of the setter & update code elsewhere
- Just manually set it to false after creating & setting the sub-menu
- Manually set it to true before ever creating the sub-menu, so that its always shown, and maybe make the icon a question mark or exclamation mark indicate that this is a verb will need confirmation?

ElectroJr reviewed (approved):
I Initially thought this would just be a confirmation dialog popup, but I think I prefer this now, because its not as much of an "interruption" when performing the action. 

Other than minor comments/issues, seems all good to me.

ShadowCommander commented:
Addressed reviews and moved more style things to StyleNano.

I made the > icon show for confirmation verbs, but made it a separate style class in StyleNano if someone wants to change it to a different icon.
